### PR TITLE
fix: ensure masked string/bytestring dtypes can hold 'nan' fill value…

### PR DESCRIPTION
**Pull request by JamesBrofos**
… in to_dataframe

When converting masked string or bytestring arrays to pandas DataFrames, narrow dtypes (e.g., U1, U2, S1, S2) may not have sufficient width to hold the fill value 'nan' (3 characters) or b'nan' (3 bytes). This caused issues during the conversion process.

This fix checks the dtype width before filling masked values and resizes the dtype to at least 3 characters/bytes when necessary, ensuring that 'nan' or b'nan' can be properly inserted for masked values.

- Check if dtype is string (U) or bytestring (S)
- Calculate character/byte width from dtype.itemsize
- Resize to minimum width of 3 if needed
- Add tests for edge cases

This PR addresses the concern in #3691

### CODE REVIEW

File: tests/test_3692_to_dataframe_masked_string_dtype_resize.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward/blob/main/LICENSE
+
+from __future__ import annotations
+
+import awkward as ak
+
+
```
Comment (ianna) [inline tests/test_3692_to_dataframe_masked_string_dtype_resize.py:11]:
```suggestion
pandas = pytest.importorskip("pandas")
```

**Review by ianna — CHANGES_REQUESTED** [review]
@JamesBrofos - This looks great! Please, check if `pandas` is installed to skip the tests when it's not. We don't have `pandas` installed in all CI workflows. Thanks!

**Comment by ianna** [comment]
@all-contributors please add @JamesBrofos for code

**Review by ianna — APPROVED** [review]
@JamesBrofos - Great! Thanks for fixing it. The tests pass, I will merge it. Thanks.
